### DISCUSSION TITLE
Settings in user_config_override.h not applied to compiled firmware

### DISCUSSION THREAD

**Discussion by dl5dla** [HASwitchPlate/openHASP#784] [General]
Hi,
I try compile the code with language set to German. For my M5Stack Core2 I did the following:

In user_config_override.h, I enabled "#define HASP_LANGUAGE de_DE", the others disabled

In platformio_override.h I enabled:
  user_setups/esp32/*.ini
  -DUSE_CONFIG_OVERRIDE
  m5stack-core2
  
  The code is compiled successfully, but after uploading the compiled firmware "build_output/firmware/m5stack-core2_ota_v0.7.0-rc12_73c1539.bin" the gui is still in English language. For test, I also set a MQTT username in user_config_override.h, which is ignored and not shown in the gui.
  
To test, whether the file user_config_override.h is really used during compilation, I added a wrong string to the file, which gave me an error (as expected). 

Is there any additional measure I have to take to get a firmware with German gui?

**Comment by fvanroie** [comment]
`#define HASP_LANGUAGE de_DE` will only change the  language of the console messages.
The web UI language changing interface isn't implemented at the moment.

If you set *preferences* in the `user_config_override.h` they are use as the *default* username and password for when no configuration is found. So if a previous configuration is present, that will be used instead of the override. To test the new default settings, erase the flash first.

**Reply by dl5dla** [reply]
Ok, that explains it. Thanks a lot.

**Comment by fvanroie** [comment]
You could try to rename `data\de_DE.json` to `data\en.json` but I haven't tried it.

**Comment by dl5dla** [comment]
Yes, that works - thank you. There are only a few strings which are hardcoded to the cpp files and therefore not able to be changed. 

I am still searching for a way to display dates including weekdays in German language ("template":"%a, %e. %b %Y"). As far as I understand, the function strftime() is used in hasp_event.cpp to translate the magic string. This function should take the localization into account. I tried to add    setlocale(LC_TIME, "de_DE")  in front of strftime(), but this did not change anything.
Any ideas how to get the German date string?
Thanks.
 Peter

**Reply by fvanroie** [reply]
I don't think `setlocale` has any impact on `strftime` in ESP-IDF. You would need to ask in that project, as it is a system function.
If you really want to have a localized `strftime`, I think you would have to [re-implement that function yourself](https://github.com/arnoldrobbins/strftime).

If you can identify the hardcoded strings, they can be added to `src\lang\en_US.h` to be translated.

**Reply by dl5dla** [reply]
Thanks. I guess for the time being I will keep the English gui. And regarding the German weekday, I can let the server push the German string once a day.
